src/Altinn.App.Core/Features/Validation/Default/LegacyFileAnalyzerValidator.cs

```diff
@@ -0,0 +1,66 @@
+namespace Altinn.App.Core.Features.Validation.Default;
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Altinn.App.Core.Features.FileAnalysis;
+using Altinn.App.Core.Models.Validation;
+using Altinn.Platform.Storage.Interface.Models;
+
+/// <summary>
+/// Implementation of <see cref="IFileUploadValidator"/> that uses the legacy 
+/// <see cref="IFileValidator"/> and <see cref="IFileAnalyser"/>.
+/// </summary>
+public class LegacyFileAnalyzerValidator : IFileUploadValidator
+{
+    private readonly IEnumerable<IFileValidator> _fileValidators;
+    private readonly IEnumerable<IFileAnalyser> _fileAnalysers;
+
+    /// <summary>
+    /// Constructor for depency injection.
+    /// </summary>
+    public LegacyFileAnalyzerValidator(IEnumerable<IFileValidator> fileValidators, IEnumerable<IFileAnalyser> fileAnalysers)
+    {
+        _fileValidators = fileValidators;
+        _fileAnalysers = fileAnalysers;
+    }
+
+    /// <inheritdoc/>
+    public string DataType => "*";
+
+    /// <inheritdoc/>
+    public async Task<List<ValidationIssue>> Validate(Instance instance, DataType dataType, byte[] fileContent, string? filename, string? mimeType, string? language)
+    {
+        var fileAnalysisResults = new List<FileAnalysisResult>();
+        if (dataType.EnabledFileAnalysers?.Count > 0)
+        {
+            // Run file analysis
+            foreach (var analyserName in dataType.EnabledFileAnalysers)
+            {
+                var analyser = _fileAnalysers.First(a => a.Id == analyserName);
```

Comment:
nit/question: throw a nicer exception as opposed to the builtin `InvalidOperationException` that `First` might throw. I.e. `throw new InvalidOperationException("Could not find file analyzer with ID: " + analyzerName)`.

My assumption is that `analyserName` comes from the app metadata JSON file. Ignore me if this is already validated on startup or something 😄

Reply:
another q for my learning purposes since I'm new: I'm not sure if this behavior is defined or communicated somewhere, but in places where we call user code, that user code migh throw exceptions. In this case this shortcircuits the whole request. Another option would be to gather exceptions in a loop and throw an `AggregateException` in the end where we end up with some results and some exceptions. Another would be to make "unexpected exception" a special type of analysis result. If this code was instrumented, we could also signal failure on spans/activities for individual analyzers for example. 

You've probably thought and decided on this stuff before, but it's helpful to know the context and design decisions made for me as a new person

Reply:
1) Yes, totally valid point. I'll add your suggestion
```suggestion
                var analyser = _fileAnalysers.FirstOrDefault(a => a.Id == analyserName)
                    ?? throw new InvalidOperationException("Could not find file analyzer with ID: " + analyzerName);
```

Reply:
I don't think we have a consistent view of how to handle exceptions in extension points. In validations I used this approach which at least logs each exception individually, even though I wanted to do them in parallel because they most likely will call external apis and waiting in sequence does not make sense. You seem to have more experience than me so please do add your suggestions.